### ISSUE TITLE
[RFC] [Validator] adding a new WordCount constraint validator

### ISSUE THREAD

**Issue by hhamon** [symfony/symfony#6813]
Hi guys,

I was thinking of adding a new constraint validator similar to the `Length` constraint but for counting words in a string. Sometimes, when writing a text or a description, we're asked to write at least x words or at most x words.

Do you think it would be needed in the validator component?

**Comment by damienalexandre** [comment]
ICU / Intl is providing a breakIterator to detect words / sentences etc: https://unicode-org.github.io/icu/userguide/boundaryanalysis/#word-boundary

So we don't have to agree on what a word is, that's not for Symfony to decide. We can leverage the BreakIterator like this:

```php
<?php

function getWords(string $content)
{
    $bi = \IntlBreakIterator::createWordInstance();
    $bi->setText($content);
    
    $words = iterator_to_array($bi->getPartsIterator());
    $words = array_map('trim', $words);
    
    return array_values(array_filter($words));
}

var_dump(getWords('次のバスは遠いです')); // Expected 6
var_dump(getWords('Ton père je suis !')); // Expected 5
```

https://3v4l.org/lv95t#v8.3.8

So I would like to suggest we reopen this issue to explore this path, maybe even add a method in the String component?

Thanks!

**Comment by manuelsucco** [comment]
Hi,

I'm new to contributing to symfony, but work with it daily. We use word count for limiting paper abstract sizes and such. I don't think this validator is too specific and with the `IntlBreakIterator`-class it seems grounded enough. There could even be a SentenceCount validator without much more work.

I offer to help.

**Comment by kriskoch** [comment]
Yeah I dont think 'too specific' can be a valid argument anymore, when we have Bic, ISBN, etc

**Comment by xabbuh** [comment]
implemented in #57716
